tasks/main.yml

```diff
@@ -14,11 +14,16 @@
 
 - name: Ensure Cockpit Web Console packages are installed.
   package:
-    name: "{{ __cockpit_packages[cockpit_packages] }}"
+    name: "{{ __cockpit_packages[cockpit_packages]
+              if cockpit_packages in __cockpit_package_types
+              else cockpit_packages }}"
     state: present
-    enablerepo: "{{ cockpit_enablerepo | default(omit, true) }}"
-    exclude: "{{ __cockpit_packages_exclude | default(omit, true) }}"
 
+- name: Remove undesired Cockpit packages
+  package:
+    name: "{{ __cockpit_packages_exclude }}"
+    state: absent
+  when: __cockpit_packages_exclude is defined
```

Comment:
@tabowling There is no platform independent way to specify `exclude_packages` - you have to use the `dnf`, `yum`, or `apt` modules to do that - this is how you do it in a platform independent way.

Reply:
Ahhh, OK.  I think that field worked for me with both the yum and dnf module.  So maybe we just move that task to the setup-DISTRO.yml file and use the platform native package manager so that we can make those exclusions?  Would that be cleaner than installing, then uninstalling?

Reply:
yes - let me work up a proposal

Reply:
@tabowling please see the latest commit

Reply:
@richm do you really need `exclude_packages`? Can't you use one of the set theory filters to remove `_cockpit_packages_exclude` from `__cockpit_packages` (specifically, `difference`)?

Reply:
hmm that won't work if the packages are specified using a wildcard (`cockpit*`).

Reply:
I think the use case for `exclude_packages` is so that I can install `cockpit*` (e.g. `dnf install cockpit*`) but blacklist certain packages.  So I can't use the set theory filters unless I can, in the playbook, expand `cockpit*` to an actual ansible `list` valued list of packages.

Reply:
> hmm that won't work if the packages are specified using a wildcard (cockpit*).

Are you talking about in the install case or in the `state: absent` case?

Reply:
the install case. I mean, my suggestion to use a set filter will not work because of this.